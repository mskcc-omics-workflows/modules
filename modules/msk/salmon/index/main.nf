process SALMON_INDEX {
    tag "$transcript_fasta"
    label "process_medium"

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/salmon:0.14.0' :
        'docker.io/mskcc/salmon:0.14.0' }"

    input:
    path transcriptome_fasta

    output:
    path "salmon"      , emit: index
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def threads = task.cpus * 2
    """
    salmon \\
        index \\
        --threads $threads \\
        -t $transcriptome_fasta \\
        ${args} \\
        -i salmon

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        salmon: \$(echo \$(salmon --version) | sed -e "s/salmon //g")
    END_VERSIONS
    """

    stub:
    """
    mkdir salmon
    touch salmon/hash.bin
    touch salmon/header.json
    touch salmon/indexing.log
    touch salmon/quasi_index.log
    touch salmon/refInfo.json
    touch salmon/rsd.bin
    touch salmon/sa.bin
    touch salmon/txpInfo.bi
    touch salmon/versionInfo.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        salmon: \$(echo \$(salmon --version) | sed -e "s/salmon //g")
    END_VERSIONS
    """
}
