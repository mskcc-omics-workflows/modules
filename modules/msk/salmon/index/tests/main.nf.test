nextflow_process {

    name "Test Process SALMON_INDEX"
    script "../main.nf"
    process "SALMON_INDEX"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "salmon"
    tag "salmon/index"

     test("rna - fastq") {

        when {
            process {
                """
                input = [
                    file(params.test_data_mskcc['rediscoverte']['rollup_part_19_fa_gz'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    file(process.out.index[0]).name
                ).match() }
            )
        }

    }

     test("rna - fastq - stub") {
        options "-stub"
        when {
            process {
                """
                    input = [file('rollup_part_19_fa_gz')]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    file(process.out.index[0]).name
                ).match() }
            )
        }

    }

}
