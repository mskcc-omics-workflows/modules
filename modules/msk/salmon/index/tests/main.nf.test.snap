{
    "rna - fastq - stub": {
        "content": [
            [
                "versions.yml:md5,3192b5ea8153bdf0e2c3139a92abcac3"
            ],
            "salmon"
        ],
        "timestamp": "2024-07-01T13:13:59.722525"
    },
    "rna - fastq": {
        "content": [
            [
                "versions.yml:md5,3192b5ea8153bdf0e2c3139a92abcac3"
            ],
            "salmon"
        ],
        "timestamp": "2024-07-01T13:13:53.299047"
    }
}